File: ELearningFrontEnd/src/app/dtos/report/report.req.ts
export class ReportReq{

    entityType: string;
    entityId: number;
    userId: number;
    reason: string;

    constructor(data: any) {
        this.entityType = data.entityType;
        this.entityId = data.entityId;
        this.userId = data.userId;
        this.reason = data.reason;
    }
}

File: ELearningBackEnd/src/main/java/com/example/elearningbackend/report/ReportReq.java
package com.example.elearningbackend.report;

import com.example.elearningbackend.common.EntityNameEnum;
import jakarta.validation.constraints.NotBlank;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@NoArgsConstructor
public class ReportReq {

    @NotBlank
    private EntityNameEnum entityType;

    private long entityId;

    private long userId;

    @NotBlank
    private String reason;

}
